Code/RDGeneral/RDValue-doublemagic.h

```diff
@@ -0,0 +1,410 @@
+//  Copyright (c) 2015, Novartis Institutes for BioMedical Research Inc.
+//  All rights reserved.
+//
+// Redistribution and use in source and binary forms, with or without
+// modification, are permitted provided that the following conditions are
+// met:
+//
+//     * Redistributions of source code must retain the above copyright
+//       notice, this list of conditions and the following disclaimer.
+//     * Redistributions in binary form must reproduce the above
+//       copyright notice, this list of conditions and the following
+//       disclaimer in the documentation and/or other materials provided
+//       with the distribution.
+//     * Neither the name of Novartis Institutes for BioMedical Research Inc.
+//       nor the names of its contributors may be used to endorse or promote
+//       products derived from this software without specific prior written
+//       permission.
+//
+// THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS
+// "AS IS" AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT
+// LIMITED TO, THE IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR
+// A PARTICULAR PURPOSE ARE DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT
+// OWNER OR CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL,
+// SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT
+// LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES; LOSS OF USE,
+// DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND ON ANY
+// THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT
+// (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE
+// OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+//
+#ifndef RDKIT_RDVALUE_PTRMAGIC_H
+#define RDKIT_RDVALUE_PTRMAGIC_H
+
+#include <stdint.h>
+#include <cassert>
+#include <boost/any.hpp>
+#include "Invariant.h"
+#include <iostream>
+#include <iomanip>
+#include <sstream>
+#include <vector>
+#include <string>
+#include <boost/utility.hpp>
+#include <boost/lexical_cast.hpp>
+#include <cmath>
+#include <boost/type_traits.hpp>
+#include <boost/static_assert.hpp>
+#include "LocaleSwitcher.h"
+#include <boost/detail/endian.hpp>
+
+#if defined(BOOST_BIG_ENDIAN) 
+BOOST_STATIC_ASSERT(false && "RDValue only has been tested under little endian cpus");
+#endif
+
+namespace RDKit {
+
+  // Inspired by
```

Comment:
The Debian distribution (which the RDKit is included in) does support big endian CPUs. So this can't be in the code. If you have concerns about supporting big endian, we should figure out a way to test there.